### PR TITLE
Poc page test helper

### PR DESCRIPTION
**Pull request by jamesbhobbs**
Resolves n/a 

**Overall change:** _A very high-level summary of easily-reproducible changes that can be understood by non-devs._

**Code changes:**

- _A bullet point list of key code changes that have been made._
- _When describing code changes, try to communicate **how** and **why** you implemented something a specific way, not just **what** has changed._

---

- [ ] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

### CODE REVIEW

File: cypress/integration/pages/errorPages/index.js [hunk 1]
```diff
@@ -1,64 +1,98 @@
 import testData from '../../../../src/app/lib/config/services';
 import config from '../../../support/config/services';
+import iterator from './pagehelper';
 
 // Do not change the config to enable these on test or live, those error pages are cached versions of what we see locally.
-const serviceHasErrorPage = service =>
+const filterCondition = service =>
```
Comment (jamesbhobbs) [inline cypress/integration/pages/errorPages/index.js:6]: Next steps would be to factor this out and pass the route to the helper

File: cypress/integration/pages/errorPages/pagehelper.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import config from '../../../support/config/services';
+
+const iterator = (filterCondition, tests, canonicalOnlyTests, ampOnlyTests) => {
+  Object.keys(config)
+    .filter(filterCondition)
+    .forEach(service => {
+      tests(service);
+      if (canonicalOnlyTests) canonicalOnlyTests(service);
+      if (ampOnlyTests) ampOnlyTests(service);
```
Comment (jamesbhobbs) [inline cypress/integration/pages/errorPages/pagehelper.js:9]: With the route passed to the helper you can append .amp automatically here

File: cypress/integration/pages/errorPages/pagehelper.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import config from '../../../support/config/services';
+
+const iterator = (filterCondition, tests, canonicalOnlyTests, ampOnlyTests) => {
+  Object.keys(config)
+    .filter(filterCondition)
+    .forEach(service => {
+      tests(service);
```
Comment (jamesbhobbs) [inline cypress/integration/pages/errorPages/pagehelper.js:7]: With route passed to the helper you can run tests twice once with route passed in and once with route.amp

File: cypress/integration/pages/errorPages/index.js (hunk 1 above)
Comment (jamesbhobbs) [reply cypress/integration/pages/errorPages/index.js:6]: Nasty object/string js work therein

File: cypress/integration/pages/errorPages/pagehelper.js (hunk 3 above)
Comment (jamesbhobbs) [reply cypress/integration/pages/errorPages/pagehelper.js:7]: also then nest the canon and amp tests into this to benefit from the same cy.visit, saving page views

**Comment by dr3** [comment]
Closing in favour of https://github.com/bbc/simorgh/pull/3053

**Pull request closed without merging**